internal/tableland/tableland.go

```diff
@@ -50,102 +23,27 @@ type TxnReceipt struct {
-	CreateStatement string `json:"create_statement"`
-}
-
-// ValidateCreateTableResponse is a ValidateCreateTable response.
-type ValidateCreateTableResponse struct {
-	StructureHash string `json:"structure_hash"`
-}
-
-// ValidateWriteQueryRequest is a ValidateWriteQuery request.
-type ValidateWriteQueryRequest struct {
-	Statement string `json:"statement"`
-}
-
-// ValidateWriteQueryResponse is a ValidateWriteQuery response.
-type ValidateWriteQueryResponse struct {
-	TableID string `json:"table_id"`
-}
-
-// SetControllerRequest is a user SetController request.
-type SetControllerRequest struct {
-	Controller string `json:"controller"`
-	TokenID    string `json:"token_id"`
-}
-
-// SetControllerResponse is a RunSQL response.
-type SetControllerResponse struct {
-	Transaction struct {
-		Hash string `json:"hash"`
-	} `json:"tx"`
-}
-
 // SQLRunner defines the run SQL interface of Tableland.
 type SQLRunner interface {
-	RunReadQuery(context.Context, RunReadQueryRequest) (RunReadQueryResponse, error)
+	RunReadQuery(context.Context, string) (interface{}, error)
 }
 
 // Tableland defines the interface of Tableland.
 type Tableland interface {
 	SQLRunner
-	ValidateCreateTable(context.Context, ValidateCreateTableRequest) (ValidateCreateTableResponse, error)
-	ValidateWriteQuery(context.Context, ValidateWriteQueryRequest) (ValidateWriteQueryResponse, error)
-	RelayWriteQuery(context.Context, RelayWriteQueryRequest) (RelayWriteQueryResponse, error)
-	GetReceipt(context.Context, GetReceiptRequest) (GetReceiptResponse, error)
-	SetController(context.Context, SetControllerRequest) (SetControllerResponse, error)
-}
-
-// TableID is the ID of a Table.
-type TableID big.Int
-
-// String returns a string representation of the TableID.
-func (tid TableID) String() string {
-	bi := (big.Int)(tid)
-	return bi.String()
-}
-
-// ToBigInt returns a *big.Int representation of the TableID.
-func (tid TableID) ToBigInt() *big.Int {
-	bi := (big.Int)(tid)
-	b := &big.Int{}
-	b.Set(&bi)
-	return b
-}
-
-// NewTableID creates a TableID from a string representation of the uint256.
-func NewTableID(strID string) (TableID, error) {
-	tableID := &big.Int{}
-	if _, ok := tableID.SetString(strID, 10); !ok {
-		return TableID{}, fmt.Errorf("parsing stringified id failed")
-	}
-	if tableID.Cmp(&big.Int{}) < 0 {
-		return TableID{}, fmt.Errorf("table id is negative")
-	}
-	return TableID(*tableID), nil
-}
-
-// NewTableIDFromInt64 returns a TableID from a int64.
-func NewTableIDFromInt64(intID int64) (TableID, error) {
-	tableID := &big.Int{}
-	tableID.SetInt64(intID)
-	return TableID(*tableID), nil
+	ValidateCreateTable(context.Context, string) (string, error)
+	ValidateWriteQuery(context.Context, string) (tables.TableID, error)
+	RelayWriteQuery(context.Context, string) (tables.Transaction, error)
+	GetReceipt(context.Context, string) (bool, *TxnReceipt, error)
+	SetController(context.Context, common.Address, tables.TableID) (tables.Transaction, error)
```

Comment:
Also, since we're not needing to provide JSON RPC conformity, we can use more fluent types like `TableID`, `Transaction`, and `TxnReceipt` in the API.